cloudformation/iam_federation.js

```diff
@@ -64,19 +64,32 @@ function addStatement(input, callback) {
   var assumeDoc = input.assumeDoc;
   var federationRoleName = input.federationRoleName;
   var lambdaRoleArn = input.roleArn;
-  var lambdaStatement = {
-    Sid: '',
-    Effect: 'Allow',
-    Principal: { AWS: lambdaRoleArn },
-    Action: 'sts:AssumeRole'
-  };
+
+  console.log(assumeDoc.Statement);
   var statements = assumeDoc.Statement.filter(function(statement) {
```

Comment:
One thing I missed is that to check 'Effect' and 'Action' of each statement for sure like 

`var statements = assumeDoc.Statement.filter(statement => statement.Effect == 'Allow' && statement.Action == 'sts:AssumeRole').filter(function(statement) {`